sdk/src/logs/batch_log_record_processor.cc

```diff
@@ -65,7 +65,7 @@ void BatchLogRecordProcessor::OnEmit(std::unique_ptr<Recordable> &&record) noexc
   {
     // signal the worker thread
     synchronization_data_->is_force_wakeup_background_worker.store(true, std::memory_order_release);
-    synchronization_data_->cv.notify_one();
+    synchronization_data_->cv.notify_all();
```

Comment:
Could there be performance impact on the switching of `notify_one` to `notify_all` in `OnEmit`?

Reply:
In my understanding, there is no difference between `notify_one` and `notify_all` when there is only a single background thread.
However, we should use `notify_all` in background threads (for example, in `BatchLogRecordProcessor::OnEmit`) to wake up all waiting threads. To keep things simple and reduce the chance of errors, it's best to consistently use `notify_all` in all cases.